subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -315,31 +315,66 @@ export function updateRevenue(
 
   const deltaReserveBalance = reserveBalance.minus(assetStatus.reserveBalance);
 
-  const deltaProtocolSideRevenue = deltaReserveBalance
+  const newProtocolSideRevenue = deltaReserveBalance
     .toBigDecimal()
     .times(market.exchangeRate!) // convert to underlying
     .div(DECIMAL_PRECISION)
     .times(token.lastPriceUSD!);
 
-  // assume reserve fee is interest revenue
-  // revenue from liquidation reserve fee is handled in rollbackRevenue()
+  // AssetStatus.reserveBalance may include protocol side revenue
+  // from interest and liquidation; split protocol side revenue
+  // if it is a liquidation
+  let newLiquidationRevenue = BIGDECIMAL_ZERO;
+  let newProtocolSideInterestRevenue = newProtocolSideRevenue;
+  let newTotalInterestRevenue = newProtocolSideInterestRevenue;
+  let newSupplySideRevenue = BIGDECIMAL_ZERO;
+  const repay = getRepayForLiquidation(event);
+  if (repay) {
+    // split reserve fee from liquidation
+    const repayAmount = bigIntToBDUseDecimals(repay, DEFAULT_DECIMALS);
+    // Line 156 Liquidation.sol: repay * (1 + 0.02)
+    const desiredRepay = repayAmount.times(BIGDECIMAL_ONE.plus(UNDERLYING_RESERVES_FEE.div(DECIMAL_PRECISION)));
```

Comment:
some more comments or different variable naming would help here.

This snippet of code covers a very nuanced section of methodology, so it will be best to be as descriptive as possible for any onlookers. And me lol, I am struggling to follow exactly what is happening and why.

Reply:
fixed. Let me know what you think of the revised comments

Reply:
I think it looks more clear! nice job